src/query/models/tag.go

```diff
@@ -85,21 +90,21 @@ func (m MatchType) String() string {
 // Matcher models the matching of a label.
 type Matcher struct {
 	Type  MatchType `json:"type"`
-	Name  string    `json:"name"`
-	Value string    `json:"value"`
+	Name  []byte    `json:"name"`
+	Value []byte    `json:"value"`
```

Comment:
Just a warning, the JSON serializer will now encode these as base64:
https://play.golang.org/p/T3Xa8OGyWML

Reply:
Do matchers also need to be bytes?  These only come through for reads yeah?

Reply:
Yeah, these are only needed for reads, but figured may as well convert them to bytes; would the base64 be a problem? We only really use it for sending to the `session` or encode them to `[]byte` anyway on remote fetches